src/Controls/src/Core/Platform/Android/TabbedPageManager.cs

```diff
@@ -277,8 +277,12 @@ void OnChildrenCollectionChanged(object sender, NotifyCollectionChangedEventArgs
 			e.Apply((o, i, c) => SetupPage((Page)o), (o, i) => TeardownPage((Page)o), Reset);
 
 			ViewPager2 pager = _viewPager;
-			var adapter = (MultiPageFragmentStateAdapter<Page>)pager.Adapter;
-			adapter.CountOverride = Element.Children.Count;
+
+			if (pager.Adapter is MultiPageFragmentStateAdapter<Page> adapter)
+			{
+				adapter.CountOverride = Element.Children.Count;
```

Comment:
Should this maybe be just an interface member so anyone can use this feature - unless it is just a very specific case?

Reply:
Yea, my plan is to do a second PR that opens up `TabbedPageManager` and `MultiPageFragmentStateAdapter` for .NET8. Plus adds some create hooks etc... Hopefully I can move them back into CORE.

This PR, which I hope to backport, is mainly to enable a user to replace the `Adapter` with a custom one and make it so the `TabbedPageManager` doesn't crash.  Which will allow users to customize or workaround any issues in .NET7 until we release .NET8 fixes